src/services/fungibleTokens.js

```diff
@@ -0,0 +1,89 @@
+import BN from 'bn.js';
+import * as nearApiJs from 'near-api-js';
+
+import sendJson from '../tmp_fetch_send_json';
+import { parseTokenAmount } from '../utils/amounts';
+import { ACCOUNT_HELPER_URL } from '../utils/wallet';
+
+const {
+    transactions: {
+        functionCall
+    },
+    utils: {
+        format: { 
+            parseNearAmount
+        }
+    }
+} = nearApiJs;
+
+const FT_MINIMUM_STORAGE_BALANCE = parseNearAmount('0.00125');
+const FT_STORAGE_DEPOSIT_GAS = parseNearAmount('0.00000000003');
+const FT_TRANSFER_GAS = parseNearAmount('0.00000000003');
+const FT_TRANSFER_DEPOSIT = '1'; // 1 yoctoNear
+
+export class FungibleTokens {
+    constructor(account) {
+        this.account = account;
+    }
+
+    async isStorageBalanceAvailable(contractName, accountId) {
+        return new BN((await this.getStorageBalance(contractName, accountId)).total).gte(new BN(FT_MINIMUM_STORAGE_BALANCE));
+    }
+
+    async getStorageBalance(contractName, accountId) {
+        return await this.account.viewFunction(contractName, 'storage_balance_of', { account_id: accountId });
+    }
+
+    async transfer(contractName, amount, memo, receiver, contractMetadata) {
+        return await this.signAndSendTransaction(contractName, [
+            functionCall('ft_transfer', {
+                receiver_id: receiver,
+                amount: await this.calculateAmount(contractName, amount, contractMetadata),
+                memo: memo
+            }, FT_TRANSFER_GAS, FT_TRANSFER_DEPOSIT)
+        ]);
+    }
+
+    async calculateAmount(contractName, amount, contractMetadata) {
+        if (!contractMetadata) {
+            contractMetadata = await this.getMetadata(contractName);
+        }
+        return parseTokenAmount(amount, contractMetadata.decimals);
+    }
+
+    async transferStorageDeposit(contractName, accountId) {
+        return this.signAndSendTransaction(contractName, [
+            functionCall('storage_deposit', {
+                account_id: accountId,
+                registration_only: true,
+            }, FT_STORAGE_DEPOSIT_GAS, FT_MINIMUM_STORAGE_BALANCE)
+        ]);
+    }
+
+    async signAndSendTransaction(receiverId, actions) {
+        return await this.account.signAndSendTransaction(receiverId, actions);
+    }
+
+    getLikelyTokenContracts() {
+        return sendJson('GET', `${ACCOUNT_HELPER_URL}/account/${this.account.accountId}/likelyTokens`);
+    }
+
+    async getMetadata(contractName) {
```

Comment:
Since this _entire object_ is now our programmatic interface for interaction with contracts that adhere to our FungibleToken standard, It'd be great if we had a JSDoc _on the the class_ that links to the entire specification.

Reply:
Sure thing, let me handle it in the near future, as time is chasing us.

Reply:
I don't think we're in quite that much of a hurry -- it should only take 30 seconds; you don't even need to go find the spec, because it's the root of the link we've already got here on `getMetaData()`:
https://github.com/near/NEPs/tree/master/specs/Standards/FungibleToken

Reply:
Done. I understood that you would like to add comments to all methods with params descriptions :)

Reply:
Ah, No, we'll get to JSDoc'ing later I think :D :) thanks!

Reply:
Sure thing, it would be helpful for external contributors for sure.